File: Bitwise/Medium/LET_no260_Single Number III.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        diff = 0
        for n in nums: 
            diff ^= n
        diff &= -diff
        res = [0, 0]
        for n in nums:
            if diff & n:
                res[0] ^= n
            else:
                res[1] ^= n
        return res

File: Bitwise/Medium/LET_no260_Single Number III.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        long diff = 0;
        for (int n : nums)
            diff ^= n;
        diff &= -diff;
        vector<int> res(2, 0);
        for (int n : nums)
            diff & n ? res[0] ^= n : res[1] ^= n;
        return res;
    }
};